invenio_rdm_records/services/services.py

```diff
@@ -424,13 +379,18 @@ def search_drafts(
     #
     # Base methods, extended with handling of deleted records
     #
-    def read(self, identity, id_, expand=False, with_deleted=False):
+    def read(self, identity, id_, expand=False, include_deleted=False):
         """Retrieve a record."""
         record = self.record_cls.pid.resolve(id_)
         result = super().read(identity, id_, expand=expand)
 
-        if record.deletion_status.is_deleted and not with_deleted:
-            raise RecordDeletedException(record, result_item=result)
+        if not include_deleted and record.deletion_status.is_deleted:
+            can_read_deleted = self.check_permission(
+                identity, "read_deleted", record=record
+            )
+
+            if not can_read_deleted:
+                raise RecordDeletedException(record, result_item=result)
```

Comment:
Major: If I am not mistaken the permission should be checked when trying to read deleted records.

```suggestion
        if not include_deleted and record.deletion_status.is_deleted:
                raise RecordDeletedException(record, result_item=result)
        if include_deleted and record.deletion_status.is_deleted:
                can_read_deleted = self.check_permission(
                     identity, "read_deleted", record=record
                 )
            if not can_read_deleted:
                 raise RecordDeletedException(record, result_item=result)       
```